.github/workflows/integrate.yaml

```
@@ -21,42 +21,63 @@ jobs:
 
   lint:
     name: Lint Code
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-24.04
     strategy:
       matrix:
         charm: [controller]
     steps:
     - uses: actions/checkout@v3
-    - run: sudo apt update && sudo apt install tox
+
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v5.3.0
+      with:
+        python-version: 3.8
+
+    - name: Install dependencies
+      run: python3 -m pip install tox
```

nitpick: Equivalent, but for consistency with our other repos:
```suggestion
      run: pip install tox
```